.github/workflows/connect-integration.yaml

```diff
@@ -0,0 +1,56 @@
+name: Connect Integration Tests
+on:
+  push:
+    branches: [main, master]
+  pull_request:
+    branches: [main, master]
+
+jobs:
+  connect-integration:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - "preview"   # Nightly builds of Connect
+          - "release"   # special value that always points to the latest Connect release
+          - "2025.09.0" # jammy
+          - "2025.03.0" # jammy (every 6 months, just bc every release would be overkill)
+          - "2024.09.0" # jammy
+          # These versions use R 4.2 and seem to take forever to install R dependencies
+          # - "2024.03.0" # jammy
+          # - "2023.09.0" # jammy
+          # These also use R 4.2 but seem fine, maybe because they are bionic?
+          - "2023.03.0" # bionic
+          - "2022.10.0" # bionic (the oldest we can test using with-connect)
+    name: Connect ${{ matrix.version }}
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+
+    steps:
+      - uses: actions/checkout@v6
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          # Only install what we need for this test, which is not all of Suggests
+          dependencies: '"hard"'
+          extra-packages: |
+            local::.
+            any::testthat
+            any::shiny
```

Comment:
Here's an example where having a dev dependency description would be lovely. 🫠 But yeah this is right for installing rsconnect, testthat, and shiny (though I did need to go look up that `local::.` and `dependencies: '"hard"'` mean that)